performance/main.py

```diff
@@ -250,6 +250,74 @@ class ArrayGOPerfREF(ArrayGOPerf):
     entry = staticmethod(ArrayGOREF)
 
 
+#-------------------------------------------------------------------------------
+class IsNaElementPerf(Perf):
+    NUMBER = 1000
+
+    def pre(self):
+        class FloatSubclass(float): pass
+
+        self.values = [
+                # Na-elements
+                np.datetime64('NaT'), np.timedelta64('NaT'), None, float('NaN'), -float('NaN'),
+
+                # Non-float, Non-na elements
+                1, 'str', np.datetime64('2020-12-31'), datetime.date(2020, 12, 31), False,
+        ]
+
+        nan = np.nan
+        nanjs = [
+                complex(nan, 0),
+                -complex(nan, 0),
+                complex(-nan, 0),
+                -complex(-nan, 0),
+                complex(0, nan),
+                -complex(0, nan),
+                complex(0, -nan),
+                -complex(0, -nan),
+        ]
+
+        float_classes = [float, np.float16, np.float32, np.float64, FloatSubclass]
+        if hasattr(np, 'float128'):
+            float_classes.append(np.float128)
+
+        cfloat_classes = [complex, np.complex64, np.complex128]
+        if hasattr(np, 'complex256'):
+            cfloat_classes.append(np.complex256)
+
+        # Append all the different types of nans across dtypes
+        for ctor in float_classes:
+            self.values.append(ctor(nan))
+            self.values.append(ctor(-nan))
+
+        for ctor in cfloat_classes:
+            for nanj in nanjs:
+                self.values.append(ctor(nanj))
+
+        # Append a wide range of float values, with different precision, across types
+        for val in (
+                1e-1000, 1e-309, 1e-39, 1e-16, 1e-5, 0.1, 0., 1.0, 1e5, 1e16, 1e39, 1e309, 1e1000,
+            ):
+            for sign in (1, -1):
+                for ctor in float_classes:
+                    self.values.append(ctor(sign * val))
+
+                for ctor in cfloat_classes:
+                    self.values.append(ctor(complex(sign * val, val)))
+                    self.values.append(ctor(complex(val, sign * val)))
```

Comment:
Maybe something like this is better? Your current code appends duplicate complex values when `sign` is `1`.

```suggestion
            for ctor in float_classes:
                self.values.append(ctor(val))
                self.values.append(ctor(-val))

            for ctor in cfloat_classes:
                self.values.append(ctor(complex(val, val)))
                self.values.append(ctor(complex(-val, val)))
                self.values.append(ctor(complex(val, -val)))
                self.values.append(ctor(complex(-val, -val)))
```